pkg/kube/inject/inject.go

```diff
@@ -649,24 +655,32 @@ func IntoObject(sidecarTemplate Templates, valuesConfig string, revision string,
 		ObjectMeta: *metadata,
 		Spec:       *podSpec,
 	}
-	if !injectRequired(ignoredNamespaces, &Config{Policy: InjectionPolicyEnabled}, &pod.Spec, pod.ObjectMeta) {
-		warningHandler(fmt.Sprintf("===> Skipping injection because %q has sidecar injection disabled\n", name))
-		return out, nil
+
+	var patchBytes []byte
+	var err error
+	if injector != nil {
+		patchBytes, err = injector.Inject(pod)
+	}
+	if patchBytes == nil {
```

Comment:
I was expecting to only do this if `injector == nil`. Is there a reason we will run it when patchBytes == nil?

Reply:
yes, as `e := &ExternalInjector{nil, nil}` in setUpExternalInjector